content-services/content-sources-backend
# Fixes 4926: job to create latest snap dist for existing repos

## Summary
A job to create the latest snapshot distribution for existing repos, because existing ones would have been missed by this PR: https://github.com/content-services/content-sources-backend/pull/844#discussion_r1815501837

## Testing steps
1. Apply this patch `git apply mypatch.patch` to prevent the latest snapshot distribution being created in the snapshot task.
```
diff --git a/pkg/tasks/snapshot_helper.go b/pkg/tasks/snapshot_helper.go
index 1c0e812..75e9bc6 100644
--- a/pkg/tasks/snapshot_helper.go
+++ b/pkg/tasks/snapshot_helper.go
@@ -63,12 +63,12 @@ func (sh *SnapshotHelper) Run(versionHref string) error {
                return err
        }
 
-       latestPathIdent := helpers.GetLatestRepoDistPath(*sh.payload.GetSnapshotIdent())
-
-       _, _, err = helper.CreateOrUpdateDistribution(sh.orgId, sh.repo.UUID, latestPathIdent, publicationHref)
-       if err != nil {
-               return err
-       }
+       //latestPathIdent := helpers.GetLatestRepoDistPath(*sh.payload.GetSnapshotIdent())
+       //
+       //_, _, err = helper.CreateOrUpdateDistribution(sh.orgId, sh.repo.UUID, latestPathIdent, publicationHref)
+       //if err != nil {
+       //      return err
+       //}
        version, err := sh.pulpClient.GetRpmRepositoryVersion(sh.ctx, versionHref)
        if err != nil {
                return err
```
2. Create multiple repos for multiple orgs, some with and without snapshotting
3. Notice that the for the "latest_snapshot_url" returned by those repos, the URL gives 404.
4. Run the job: `go run cmd/create_latest_distributions/main.go`
5. Those URLs that were giving 404s should now serve the latest snapshot
